hibernate-core/src/main/java/org/hibernate/generator/Generator.java

```diff
@@ -48,24 +49,19 @@
  * @since 6.2
  */
 public interface Generator extends Serializable {
-	/**
-	 * Specifies that the property value is generated:
-	 * <ul>
-	 * <li>{@linkplain GenerationTiming#INSERT when the entity is inserted},
-	 * <li>{@linkplain GenerationTiming#UPDATE when the entity is updated},
-	 * <li>{@linkplain GenerationTiming#ALWAYS whenever the entity is inserted or updated}, or
-	 * <li>{@linkplain GenerationTiming#NEVER never}.
-	 * </ul>
-	 *
-	 * @return The {@link GenerationTiming} specifying when the value is generated.
-	 */
-	GenerationTiming getGenerationTiming();
-
 	/**
 	 * Determines if the property value is generated in Java, or by the database.
 	 *
 	 * @return {@code true} if the value is generated by the database, or false if it is
 	 *		   generated in Java using a {@link ValueGenerator}.
 	 */
 	boolean generatedByDatabase();
+
```

Comment:
VERY much dislike decomposing the enum into 2 booleans

Reply:
Well, I guess this depends whether we really, actually support UPDATE.  I believe the truly valid options are (1) never, (2) on insert and (3) on both insert and update.

If we do in fact support just UPDATE, then my main concern here is gone.

Reply:
Yeah, it was introduced in https://hibernate.atlassian.net/browse/HHH-15672.

Reply:
@sebersole your comments have got me thinking. I wonder if, for the sake of future extensibility, and elegance, it would be better if we introduced something like:

```
enum GenerationEventType {
     INSERT, UPDATE
}
```

and then used `EntityEventType[]` to express these things.

This would follow the pattern established by `CascadeType`. So "always" would be `{INSERT,UPDATE}` and "never" would be `{}`.

Reply:
I have just implemented that, so you can see what I mean. I think it worked out well, much better than the "two booleans" thing. And I always wanted to use `EnumSet` for something ;-)

Reply:
Introducing `EventType` also made the mapping to version generation cleaner. Nice.